src/Passes/Source/AllocationManager/AllocationManager.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/IAllocationManager.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+    /// AllocationManager is a simple qubit and results allocator that can be used at compile-time.
+    /// It is based on an assumption that all qubit allocating function calls are inlined and that
+    /// qubits/results can be allocated with strictly growing IDs.
+    class BasicAllocationManager : public IAllocationManager
+    {
+      public:
+        /// Defines a named register/memory segment with start
+        /// position, end position and size.
+        struct MemoryMapping
+        {
+            String  name{""}; ///< Name of the segment, if any given
+            Index   index{0}; ///< Index of the allocation
+            Index   size{0};  ///< Size of memory segment
+            Address start{0}; ///< Start index of memory segment
+            Address end{0};   ///< Index not included in memory segment
```

Comment:
Would be nice to see a few words explaining what the "segment" is in the fragments "memory segment" (or to get "segment" replaced with something more clear). Intuitively I feel that you meant to say something like "allocated memory block" (rather than something like "code segment", "data segment", "interrupt table segment").

Reply:
I've attempted to make things more clear.